### ISSUE TITLE
Bundle the Sentry iOS SDK

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-xamarin#62]
When we bundle the [Sentry iOS SDK](https://github.com/getsentry/sentry-cocoa) we need to take into account that Mono uses some signals to interpret `NullReferenceException`s and need special case (run before other crash reports do):

See: https://www.mono-project.com/docs/advanced/signals/#incomplete-solution
See also: https://github.com/dotnet/runtime/issues/44736

This is probably something we can do in the binding project, before calling `Init` in the iOS SDK.

Proposed solution, based on the Mono docs:

```csharp
try {
} finally {
    Mono.Runtime.RemoveSignalHandlers ();
    try {
        // Init the Sentry iOS SDK:
        SentrySDK.start(...);
    } finally {
        Mono.Runtime.InstallSignalHandlers ();
    }
}
```

> Since Mono’s signal handlers are reinstalled and not restored, Mono will now chain to the signal handlers installed by EnableCrashReporting.

> The code is executed in a finally block, so that the Mono runtime will never abort it under any circumstance.

> It’s recommended to do this as early as possible when launching the process, in particular before starting any secondary threads. If any signals are raised between removing and reinstalling Mono’s signal handlers (this includes NullReferenceExceptions), the app will crash.

See: https://www.mono-project.com/docs/advanced/signals/#complete-solution

**Comment by bruno-garcia** [comment]
Superseded by: https://github.com/getsentry/sentry-dotnet/issues/1282
